src/axum/serde_json.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{DeserializeError, Deserr, JsonError};
+use axum::async_trait;
+use axum::body::HttpBody;
+use axum::extract::rejection::JsonRejection;
+use axum::extract::FromRequest;
+use axum::response::IntoResponse;
+use axum::{BoxError, Json};
+use http::{Request, StatusCode};
+use std::marker::PhantomData;
+
+/// Extractor for typed data from Json request payloads
+/// deserialised by deserr.
+///
+/// ## Extractor
+/// To extract typed data from a request body, the inner type `T` must implement the
+/// [`deserr::Deserr<E>`] trait. The inner type `E` must implement the
+/// [`DeserializeError`] trait.
+///
+/// ## Response
+/// [`axum::IntoResponse`] is implemented for any `AxumJson<T, E>`
+/// where `T` implement [`serde::Serialize`].
+#[derive(Debug)]
+pub struct AxumJson<T, E>(pub T, PhantomData<E>);
+
+#[derive(Debug)]
+pub enum AxumJsonRejection {
+    DeserrError(JsonError),
+    JsonRejection(JsonRejection),
+}
```

Comment:
We should let the end user decide which kind of error they want to return.

```suggestion
#[derive(Debug)]
pub enum AxumJsonRejection<E: DeserializeError> {
    DeserrError(E),
    JsonRejection(JsonRejection),
}
```